modules/data_release/jsx/uploadFileForm.js

```diff
@@ -0,0 +1,230 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import ProgressBar from 'ProgressBar';
+
+
+/**
+ * Upload File Form
+ *
+ * Module component rendering the upload file form modal window
+ */
+class UploadFileForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: {},
+      hasError: {},
+      isLoaded: false,
+      uploadProgress: -1,
+    };
+
+    this.updateFormElement = this.updateFormElement.bind(this);
+    this.validateAndSubmit = this.validateAndSubmit.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+  }
+
+  componentDidMount() {
+    let self = this;
+    fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then( (resp) => resp.json())
+      .then( (data) => self.setState({data: data, isLoaded: true}))
+      .catch( (error) => {
+        self.setState({error: 'An error occurred when loading the form!'});
+        console.error(error);
+      });
+  }
+
+  render() {
+    // Data loading error
+    if (this.state.error !== undefined) {
+      return (
+        <div className='alert alert-danger text-center'>
+          <strong>
+            {this.state.error}
+          </strong>
+        </div>
+      );
+    }
+
+    return (
+      <FormElement
+        name='uploadfile'
+        fileUpload={true}
+        onSubmit={this.validateAndSubmit}>
+        <FileElement
+          name='file'
+          label='File to upload'
+          onUserInput={this.updateFormElement}
+          required={true}
+          value={this.state.formData.file}
+        />
+        <TextboxElement
+          name='version'
+          label='Version'
+          onUserInput={this.updateFormElement}
+          required={false}
+          value={this.state.formData.version}
+        />
+        <ButtonElement label='Upload File'/>
+        <div className='row'>
+          <div className='col-sm-9 col-sm-offset-3'>
+            <ProgressBar value={this.state.uploadProgress}/>
+          </div>
+        </div>
+      </FormElement>
+    );
+  }
+
+  /**
+   * Update a form Element
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding element
+   */
+  updateFormElement(formElement, value) {
+    let formData = this.state.formData;
+
+    if (value === '') {
+      formData[formElement] = null;
+    } else {
+      formData[formElement] = value;
+    }
+
+    this.setState({formData: formData});
+  }
+
+  /**
+   * Validate and submit the upload
+   */
+  validateAndSubmit() {
```

Comment:
It would be good to upgrade this function (or the uploadFile function) to handle the error cases in `ajax/FileUpload.php`.

For example I tried uploading a file when the directory didn't exist and I got a 500 error/blank page.

Instead it would be good to have the ajax script send error messages here to be displayed by the SWAL.

Reply:
actually, this is happening in `uploadFile` function. The only thing I don't get is why the error message do not show in the swal window when the file is overwritten but it shows for a new file not already uploaded...

Reply:
@johnsaigle I fixed it. Don't ask me the details on how I fixed it, I have no idea, lol!! At least now the message shows in the swal window:
![Screen Shot 2020-02-10 at 2 20 29 PM](https://user-images.githubusercontent.com/1402456/74182161-8737f300-4c10-11ea-8604-1f81aa8ae297.png)


@HenriRabalais @driusan forget my earlier comment on this PR. Looks like I figured out how to fix this somehow...